cmd/lifecycle/cli/command.go

```diff
@@ -54,6 +55,13 @@ func Run(c Command, asSubcommand bool) {
 	if err := cmd.SetLogLevel(logLevel); err != nil {
 		cmd.Exit(err)
 	}
+
+	// Warn when CNB_PLATFORM_API is unset
```

Comment:
With this placement we don't print the warning if we're just running `lifecycle -version`